transform/startf/transform.go

```diff
@@ -168,229 +144,194 @@ func ExecScript(ctx context.Context, target *dataset.Dataset, opts ...func(o *Ex
 	resolve.AllowSet = o.AllowSet
 	resolve.AllowLambda = o.AllowLambda
 	resolve.AllowNestedDef = o.AllowNestedDef
+	resolve.LoadBindsGlobally = true
+	resolve.AllowRecursion = true
```

Comment:
Do we need AllowRecursion? Enabling that is equivalent to enabling unbounded loops (while loops) which makes analysis much more complicated in the general case. Would love if we could keep this disabled.

Reply:
woah I completely forgot how hard this makes things on the static analysis side. Yes will definitely disable